Sources/Sampler/metropolis_hastings.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Simons Foundation, Inc. - All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCES_SAMPLER_METROPOLISHASTINGS_HPP
+#define SOURCES_SAMPLER_METROPOLISHASTINGS_HPP
+
+#include <Eigen/Core>
+#include <functional>
+#include <limits>
+#include <vector>
+
+#include "Sampler/abstract_sampler.hpp"
+#include "Utils/exceptions.hpp"
+#include "Utils/random_utils.hpp"
+
+namespace netket {
+
+class MetropolisHastings : public AbstractSampler {
+ public:
+  using TransitionKernel = std::function<void(
+      Eigen::Ref<const RowMatrix<double>>, Eigen::Ref<RowMatrix<double>>,
+      Eigen::Ref<Eigen::ArrayXd>)>;
```

Comment:
Since `TransitionKernel` is a `std::function` (i.e. cannot be inlined), all transition kernels should be implemented in `.cc` files.